### ISSUE TITLE
Install and configure telegraf in azure VMSS for custom metric

### ISSUE THREAD

**Issue by anildesai61** [influxdata/telegraf#14300]
### Please direct all support questsions to slack or the forums. Thank you.

Hello Team,
i have ubuntu 18.04 LTS VMSS on azure , below are the steps which i followed:

apt install telegraf -y
telegraf --input-filter cpu:mem:diskio:net --output-filter azure_monitor config > /etc/telegraf/telegraf.conf
systemctl restart telegraf

**added a line in /etc/telegraf/telegraf.conf file as**:
_resoure_id = "/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Compute/virtualMachineScaleSets/%s"_

But still telegraf metric not able to visible in the metric option at vmss, can you please help here it will be more helpful.

**telegraf status**
telegraf.service - Telegraf
Loaded: loaded (/lib/systemd/system/telegraf.service; enabled; vendor preset: enabled)
Active: active (running) since Tue 2023-11-14 18:01:09 UTC; 2min 6s ago
Docs: https://github.com/influxdata/telegraf
Main PID: 3505 (telegraf)
Tasks: 7 (limit: 4915)
CGroup: /system.slice/telegraf.service
└─3505 /usr/bin/telegraf -config /etc/telegraf/telegraf.conf -config-directory /etc/telegraf/telegraf.d
Nov 14 18:01:09 waf-teleg000000 telegraf[3505]: 2023-11-14T18:01:09Z I! Tags enabled: host=waf-teleg000000
Nov 14 18:01:09 waf-teleg000000 systemd[1]: Started Telegraf.
Nov 14 18:01:09 waf-teleg000000 telegraf[3505]: 2023-11-14T18:01:09Z I! [agent] Config: Interval:10s, Quiet:false, Hostname:"waf-teleg000000", Flush Interval:10s
Nov 14 18:02:09 waf-teleg000000 telegraf[3505]: 2023-11-14T18:02:09Z E! [agent] Error writing to outputs.azure_monitor: unable to fetch authentication credentials: azure.BearerAuthorizer#WithAuthorization: Failed to refresh the Token for request to https://centralindia.monitoring.azure.com/subscriptions/xxxx-xx-xx-xx-xxx-xxx-xxx-xxx-xxxx/resourceGroups/TELEGRAPH/providers/Microsoft.Compute/virtualMachineScaleSets/waf-telegraph1/metrics: StatusCode=400 -- Original Error: adal: Refresh request failed. Status Code = '400'. Response body: {"error":"invalid_request","error_description":"Identity not found"} Endpoint http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&resource=https%3A%2F%2Fmonitoring.azure.com%2F

**Telegraf config file.**
#Send aggregate metrics to Azure Monitor
[[outputs.azure_monitor]]
Timeout for HTTP writes.
timeout = "20s"
#Set the namespace prefix, defaults to "Telegraf/".
namespace_prefix = "Telegraf/Apache"
#Azure Monitor doesn't have a string value type, so convert string
#fields to dimensions (a.k.a. tags) if enabled. Azure Monitor allows
#a maximum of 10 dimensions so Telegraf will only send the first 10
#alphanumeric dimensions.
strings_as_dimensions = false
#Both region and resource_id must be set or be available via the
#Instance Metadata service on Azure Virtual Machines.
#Azure Region to publish metrics against
region = "centralindia"
The Azure Resource ID against which metric will be logged, e.g.
resource_id = "/subscriptions/xxx-xx-x-xxxxx-xxx-xx/resourceGroups/TELEGRAPH/providers/Microsoft.Compute/virtualMachineScaleSets/waf-telegraph1"

**Please share any example telegraf config file for vmss, i wanted to achieve based on the Apache requests in telegraf metric wanted to scale up azure vmss**

**Comment by anildesai61** [comment]
> I see you are getting support on the forums, so I am going to closet his issue.

Not exactly, can you please keep open the issue, so that can get some good suggestions and resolutions quickly?

**Comment by powersj** [comment]
The issues here are not for support.

**Comment by anildesai61** [comment]
got it @powersj  Thanks !!
